### DISCUSSION TITLE
Get the game's pause event

### DISCUSSION THREAD

**Discussion by vincecarterhu** [scripthookvdotnet/scripthookvdotnet#1315] [Q&A]
I need to monitor and do some operations while the game is paused。

Game.IsPaused always return false when game is paused , is not work

**Comment by kagikn** [comment]
In short: `Game.IsPaused` **does** actually return true when the game is showing the pause menu, but you can't have tick methods called when the game showing the pause menu **and** the game is paused/frozen. I assume you are thinking `Game.IsPaused` does not work, but it's rather about when SHVDN can call tick methods.

`IS_PAUSE_MENU_ACTIVE` returns true when the game is showing the pause menu, and I confirmed this by disassembling the native function. The primary issue about this native is, while the game is paused and showing the pause menu by pushing a pause button/key or calling `ACTIVATE_FRONTEND_MENU` with the 2nd argument true (`bPauseGame`), you can't get have tick methods called by SHVDN. Well actually, it's a limitation comes from SHV (as how regular ysc scripts work I guess).
A weird fact is, you **can** have methods subscribed to `KeyDown` or `KeyUp` called. So you can actually check if the game is paused each time a key is just pressed or keeping pressed or is just released...

I think you have to modify some SHV's resources such as its manager "script" so any SHV scripts can run tick functions even when the game is paused. I can't really help you in this right away as I don't have clear idea how to achieve though.

**Reply by vincecarterhu** [reply]
thx ,guys  I used the KeyDown method
